backends/p4tools/modules/smith/targets/bmv2/target.h

```diff
@@ -13,4 +28,86 @@ class AbstractBMv2SmithTarget : public SmithTarget {
 };
 }  // namespace P4Tools::P4Smith::BMv2
 
+namespace P4Tools::P4Smith {
+
+using namespace P4::literals;
+
+class Bmv2V1ModelDeclarationGenerator : public DeclarationGenerator {
+ public:
+    explicit Bmv2V1ModelDeclarationGenerator(
+        P4Tools::P4Smith::BMv2::AbstractBMv2SmithTarget &parent)
+        : DeclarationGenerator(static_cast<P4Tools::P4Smith::SmithTarget &>(parent)) {}
```

Comment:
You shouldn't need this. Instead require the correct parameter for `Bmv2V1ModelDeclarationGenerator`